pytorch_lightning/plugins/ddp_sequential_plugin.py

```diff
@@ -0,0 +1,411 @@
+            rpc_pipe.PipeModel.trainer = trainer
+            del rpc_pipe.PipeModel.trainer.model.sequential_module
+            rpc_pipe.PipeModel.trainer.model.sequential_module = rpc_pipe.PipeModel
+            rpc_pipe.PipeModel.configure_optimizers = trainer.model.configure_optimizers
+        super().on_accelerator_exit_rpc_process(trainer)
+
+    def set_main_rpc_process(self):
+        self.main_rpc_process = torch_distrib.get_rank(group=mpu.get_pipeline_parallel_group()) == 0
+
+    def on_main_rpc_connection(self, trainer) -> None:
+        # Create pipe_module
+        model = trainer.get_model()
+        self._find_and_init_pipe_module(model)
+        if not trainer.testing:
+            torch_distrib.barrier()  # Ensure we join main process initialization
+            model.sequential_module.foreach_worker(register_optimizers, include_self=True)
+
+    def _check_arguments(self, trainer):
+        if trainer.amp_backend is not None:
+            raise MisconfigurationException(
+                'DDPSequentialPlugin is currently not supported in Automatic Mixed Precision')
+
+    def configure_ddp(
+            self,
+            model: LightningModule, device_ids: List[int]) -> DistributedDataParallel:
+        ddp_plugin = RPCPlugin(process_group=mpu.get_data_parallel_group()).configure_ddp(model, device_ids)
+        # Plugin handle backwards across processes. Currently not supported for DDP + pipe parallel
+        ddp_plugin.PREPARE_FOR_BACKWARDS = False
+        return ddp_plugin
+
+    @rank_zero_only
+    def rpc_save_model(
+            self,
+            save_model_fn,
+            last_filepath,
+            trainer,
+            pl_module) -> None:
+        model = trainer.get_model()
+        if not hasattr(model.sequential_module, "foreach_worker"):
+            return
+        current_layers = pl_module.sequential_module
+        model.sequential_module.foreach_worker(
+            save_layers_on_all_rank_zero_workers,
+            {"gpus_per_model": self.gpus_per_model},
+            include_self=True
+        )
+        pl_module.sequential_module = load_sequential_from_saved_layers(self.gpus_per_model)
+        save_model_fn(last_filepath, trainer, pl_module)
+        pl_module.sequential_module = current_layers
+
+    def worker_optimizer_step(
+            self,
+            model: LightningModule,
+            opt_idx: int,
+            *args,
+            **kwargs) -> None:
+        model.sequential_module.foreach_worker(
+            run_optimizer,
+            {"opt_idx": opt_idx, "args": args, "kwargs": kwargs},
+            include_self=False
+        )
+
+    def distributed_sampler_kwargs(self, distributed_sampler_kwargs):
+        distributed_sampler_kwargs = dict(
+            num_replicas=mpu.get_data_parallel_world_size(),
+            rank=mpu.get_data_parallel_rank(),
+        )
+        return distributed_sampler_kwargs
```

Comment:
```suggestion
        return dict(
            num_replicas=mpu.get_data_parallel_world_size(),
            rank=mpu.get_data_parallel_rank(),
        )
```

also, the function argument `distributed_sampler_kwargs` is unused. Did you mean to update it?

Reply:
It used from accelerator.